openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+changetype: modify
+add: olcPasswordHash
+olcPasswordHash: $LDAP_PASSWORD_HASH
+EOF
+    debug_execute ldapmodify "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/password_hash.ldif"
+}
+
+########################
+# OpenLDAP configure Access Logging
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_enable_accesslog() {
+    info "Configure Access Logging"
+    # Add indexes
+    cat > "${LDAP_SHARE_DIR}/accesslog_add_indexes.ldif" << EOF
+dn: olcDatabase={2}mdb,cn=config
+changetype: modify
+add: olcDbIndex
+olcDbIndex: entryCSN eq
+-
+add: olcDbIndex
+olcDbIndex: entryUUID eq
+EOF
+    debug_execute ldapmodify "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/accesslog_add_indexes.ldif"
+    # Load module
+    ldap_load_module "${LDAP_BASE_DIR}/lib/openldap" "accesslog.so"
+    # Create AccessLog database
+    cat > "${LDAP_SHARE_DIR}/accesslog_create_accesslog_database.ldif" << EOF
+dn: olcDatabase={3}mdb,cn=config
+objectClass: olcDatabaseConfig
+objectClass: olcMdbConfig
+olcDatabase: {3}mdb
+olcDbDirectory: $LDAP_ACCESSLOG_DATA_DIR
+olcSuffix: $LDAP_ACCESSLOG_DB
+olcRootDN: $LDAP_ACCESSLOG_ADMIN_DN
+olcRootPW: $LDAP_ENCRYPTED_ACCESSLOG_ADMIN_PASSWORD
+olcDbIndex: default eq
+olcDbIndex: entryCSN,objectClass,reqEnd,reqResult,reqStart
+EOF
+    mkdir "${LDAP_ACCESSLOG_DATA_DIR}"
+    debug_execute ldapadd "${slapd_debug_args[@]}" -Q -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/accesslog_create_accesslog_database.ldif"
+    # Add AccessLog overlay
+    cat > "${LDAP_SHARE_DIR}/accesslog_create_overlay_configuration.ldif" << EOF
+dn: olcOverlay=accesslog,olcDatabase={2}mdb,cn=config
+objectClass: olcOverlayConfig
+objectClass: olcAccessLogConfig
+olcOverlay: accesslog
+olcAccessLogDB: $LDAP_ACCESSLOG_DB
+olcAccessLogOps: $LDAP_ACCESSLOG_LOGOPS
+olcAccessLogSuccess: $LDAP_ACCESSLOG_LOGSUCCESS
+olcAccessLogPurge: $LDAP_ACCESSLOG_LOGPURGE
+olcAccessLogOld: $LDAP_ACCESSLOG_LOGOLD
+olcAccessLogOldAttr: $LDAP_ACCESSLOG_LOGOLDATTR
+EOF
+    info "adding accesslog_create_overlay_configuration.ldif"
+    debug_execute ldapadd "${slapd_debug_args[@]}" -Q -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/accesslog_create_overlay_configuration.ldif"
+}
+
+########################
+# OpenLDAP configure Sync Provider
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_enable_syncprov() {
+    info "Configure Sync Provider"
+    # Load module
+    ldap_load_module "${LDAP_BASE_DIR}/lib/openldap" "syncprov.so"
+    # Add Sync Provider overlay
+    cat > "${LDAP_SHARE_DIR}/syncprov_create_overlay_configuration.ldif" << EOF
+dn: olcOverlay=syncprov,olcDatabase={2}mdb,cn=config
+objectClass: olcOverlayConfig
+objectClass: olcSyncProvConfig
+olcOverlay: syncprov
+olcSpCheckpoint: $LDAP_SYNCPROV_CHECKPPOINT
+olcSpSessionLog: $LDAP_SYNCPROV_SESSIONLOG
```

Comment:
2.6 now has olcSpSessionlogSource which is more efficient than olcSpSessionlog (obviously you'd need to have accesslog configured to use it)

Reply:
I now conditionally enable `olcSpSessionlogSource` when `LDAP_ENABLE_ACCESSLOG` is true.